pallets/liquid-staking/src/lib.rs

```diff
@@ -708,13 +737,22 @@ pub mod pallet {
                     return Err(Error::<T>::NotWithdrawn.into());
                 }
 
-                T::Assets::transfer(
-                    Self::staking_currency()?,
-                    &Self::account_id(),
-                    &who,
-                    amount,
-                    false,
-                )?;
+                if who == Self::loans_account_id() {
+                    let account_borrows =
+                        T::Loans::get_current_borrow_balance(&account_id, staking_currency)?;
+                    T::Loans::do_repay_borrow(
+                        &account_id,
+                        staking_currency,
+                        min(account_borrows, amount),
+                    )?;
+                    let redeem_amount = T::Loans::get_market_info(collateral_currency)?
+                        .collateral_factor
+                        .saturating_reciprocal_mul_ceil(amount);
```

Comment:
should be `min(account_borrows, amount)` here?

Reply:
it's already `min(account_borrows, amount)` I guess

Reply:
https://github.com/parallel-finance/parallel/blob/1a2c4933adf9b1c33356972aaa6c6c97b54d8b06/pallets/liquid-staking/src/lib.rs#L743-L751

`redeem_amount`  here is still calculated base on amount, will it be over redeemed?

Reply:
I think it shouldn't because `mint_amount` in `do_loans_fast_unstake` is calculated using `liquid_amount * exchangeRate / collateral_factor` but the item in unlockings is `liquid_amount * exchangeRate`